### DISCUSSION TITLE
Object vs. object error for maps

### DISCUSSION THREAD

**Discussion by kungfooman** [eslint/eslint#14235] [Q&A]
In order to create typed maps for TypeScript declaration files, JSDoc supports the e.g. `Object<string, numer>` syntax, which generates TypeScript declaration code like this:

![image](https://user-images.githubusercontent.com/5236548/111774139-5adcb480-88af-11eb-8d19-8f4b3906a401.png)

In VSCode this is common code that works:

![image](https://user-images.githubusercontent.com/5236548/111773983-236e0800-88af-11eb-8113-6f26c28db3e6.png)

But ESLint is throwing errors and the suggested change (preferring lower case `object`) breaks VSCode/TypeScript:

![image](https://user-images.githubusercontent.com/5236548/111773787-ea359800-88ae-11eb-912c-9a6b22a4af0b.png)

Anyone knows a proper solution for this?

**Comment by mdjermanovic** [comment]
Hi @kungfooman!

> But ESLint is throwing errors and the suggested change (preferring lower case object) breaks VSCode/TypeScript 

Can you share those errors?

**Reply by kungfooman** [reply]
Hi @mdjermanovic! This is the error from ESLint: 
![image](https://user-images.githubusercontent.com/5236548/111863504-47d3ee00-895c-11eb-8d91-f12fd71cce27.png)
While the upper case `Object<...>` syntax is what only works in VSCode/TypeScript

**Comment by mdjermanovic** [comment]
The error is reported by a third-party rule `jsdoc/check-types`.

You can open an issue in the plugin's repository:

https://github.com/gajus/eslint-plugin-jsdoc
